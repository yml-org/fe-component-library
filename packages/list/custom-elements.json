{
  "schemaVersion": "1.0.0",
  "readme": "",
  "modules": [
    {
      "kind": "javascript-module",
      "path": "list.component.ts",
      "declarations": [],
      "exports": []
    },
    {
      "kind": "javascript-module",
      "path": "src/list/list.component.ts",
      "declarations": [
        {
          "kind": "class",
          "description": "",
          "name": "ListComponent",
          "members": [
            {
              "kind": "field",
              "name": "listType",
              "type": {
                "text": "ListType | undefined"
              },
              "attribute": "listType",
              "reflects": true
            },
            {
              "kind": "field",
              "name": "listItems",
              "type": {
                "text": "ListItemType[] | undefined"
              },
              "default": "[]",
              "attribute": "listItems",
              "reflects": true
            },
            {
              "kind": "field",
              "name": "overrideDefaultListStyles",
              "type": {
                "text": "boolean | undefined"
              },
              "default": "false",
              "attribute": "overrideDefaultListStyles",
              "reflects": true
            },
            {
              "kind": "field",
              "name": "showBorder",
              "type": {
                "text": "boolean | undefined"
              },
              "default": "false",
              "attribute": "showBorder",
              "reflects": true
            },
            {
              "kind": "field",
              "name": "borderPosition",
              "type": {
                "text": "BorderPosition | undefined"
              },
              "attribute": "borderPosition",
              "reflects": true
            },
            {
              "kind": "field",
              "name": "borderStyle",
              "type": {
                "text": "BorderStyle | undefined"
              },
              "attribute": "borderStyle",
              "reflects": true
            },
            {
              "kind": "field",
              "name": "containerPartName",
              "type": {
                "text": "string | undefined"
              },
              "default": "'webcl-list-container'",
              "attribute": "containerPartName",
              "reflects": true
            },
            {
              "kind": "field",
              "name": "listPartName",
              "type": {
                "text": "string | undefined"
              },
              "default": "'webcl-list'",
              "attribute": "listPartName",
              "reflects": true
            },
            {
              "kind": "method",
              "name": "renderListItemContent",
              "privacy": "protected",
              "parameters": [
                {
                  "name": "rightSlot"
                },
                {
                  "name": "leftSlot"
                },
                {
                  "name": "listLabel"
                },
                {
                  "name": "listLabelPartName"
                }
              ]
            },
            {
              "kind": "method",
              "name": "renderAnchorOrButton",
              "privacy": "protected",
              "parameters": [
                {
                  "name": "{\n    id,\n    isAnchor,\n    isButton,\n    href,\n    listLabel,\n    rightSlot,\n    leftSlot,\n    btnClickHandler,\n    contentPartName,\n    listLabelPartName,\n  }",
                  "type": {
                    "text": "ListItemType"
                  }
                }
              ]
            },
            {
              "kind": "method",
              "name": "renderBorderStyles",
              "privacy": "protected"
            },
            {
              "kind": "method",
              "name": "renderListItems",
              "privacy": "protected"
            },
            {
              "kind": "method",
              "name": "renderUnorderedList",
              "privacy": "protected"
            },
            {
              "kind": "method",
              "name": "renderOrderedList",
              "privacy": "protected"
            },
            {
              "kind": "method",
              "name": "renderList",
              "privacy": "protected"
            }
          ],
          "attributes": [
            {
              "name": "listType",
              "type": {
                "text": "ListType | undefined"
              },
              "fieldName": "listType"
            },
            {
              "name": "listItems",
              "type": {
                "text": "ListItemType[] | undefined"
              },
              "default": "[]",
              "fieldName": "listItems"
            },
            {
              "name": "overrideDefaultListStyles",
              "type": {
                "text": "boolean | undefined"
              },
              "default": "false",
              "fieldName": "overrideDefaultListStyles"
            },
            {
              "name": "showBorder",
              "type": {
                "text": "boolean | undefined"
              },
              "default": "false",
              "fieldName": "showBorder"
            },
            {
              "name": "borderPosition",
              "type": {
                "text": "BorderPosition | undefined"
              },
              "fieldName": "borderPosition"
            },
            {
              "name": "borderStyle",
              "type": {
                "text": "BorderStyle | undefined"
              },
              "fieldName": "borderStyle"
            },
            {
              "name": "containerPartName",
              "type": {
                "text": "string | undefined"
              },
              "default": "'webcl-list-container'",
              "fieldName": "containerPartName"
            },
            {
              "name": "listPartName",
              "type": {
                "text": "string | undefined"
              },
              "default": "'webcl-list'",
              "fieldName": "listPartName"
            }
          ],
          "mixins": [
            {
              "name": "TailwindElement",
              "module": "/src/shared/tailwind.element"
            }
          ],
          "superclass": {
            "module": "src/list/list.component.ts"
          },
          "tagName": "ymlwebcl-list",
          "customElement": true
        }
      ],
      "exports": [
        {
          "kind": "js",
          "name": "ListComponent",
          "declaration": {
            "name": "ListComponent",
            "module": "src/list/list.component.ts"
          }
        },
        {
          "kind": "custom-element-definition",
          "name": "ymlwebcl-list",
          "declaration": {
            "name": "ListComponent",
            "module": "src/list/list.component.ts"
          }
        }
      ]
    }
  ]
}
