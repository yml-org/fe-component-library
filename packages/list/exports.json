{
  ".": "./src/index.js",
  "./src/*.js": "./src/*.js",
  "./package.json": "./package.json",
  "./list.component.js": "./list.component.js"
}
