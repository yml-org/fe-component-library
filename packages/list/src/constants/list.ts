export enum ListTypes {
  Ordered = 'Ordered',
  Unordered = 'Unordered',
  Description = 'Description',
}

export enum SlotPositions {
  Start = 'Start',
  End = 'End',
}

export enum BorderPositions {
  All = 'All',
  Top = 'Top',
  Left = 'Left',
  Bottom = 'Bottom',
  Right = 'Right',
}

export enum BorderStyles {
  Solid = 'solid',
  Dashed = 'dashed',
  Dotted = 'dotted',
  Double = 'double',
  Hidden = 'hidden',
  None = 'none',
}
