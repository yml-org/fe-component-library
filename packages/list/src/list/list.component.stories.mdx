import { html } from 'lit';
import { Story, Meta, Canvas } from '@storybook/addon-docs';
import './list.component.ts';
import {
  orderedListVariant,
  unOrderedListVariant,
  unOrderedListWithBottomBorder,
} from '../constants/list.variants.ts';

<Meta title="Components/List" />

# List Component

The List component is a Lit component that renders a list component that is framework independent and utilizes Tailwind CSS internally.

# Props

| Prop Name                 | Type           | Description                                                                 |
| ------------------------- | -------------- | --------------------------------------------------------------------------- |
| listType                  | ListType       | The type of list to render                                                  |
| listItems                 | ListItemType[] | The list items                                                              |
| overrideDefaultListStyles | boolean        | Flag to determine whether or not the default list styles need to be applied |
| showBorder                | boolean        | Flag to determine whether or not to show border on list elements            |
| borderPosition            | BorderPosition | Where to show the border                                                    |
| borderStyle               | BorderStyle    | The border style                                                            |
| containerPartName         | string         | The part name for the container                                             |
| listPartName              | string         | The part name for the list                                                  |

The different values for the list types are as follows:

1. Ordered
2. Unordered

The different values for the border position are as follows:

1. All
2. Top
3. Left
4. Bottom
5. Right

The different values for the border style are as follows:

1.  solid
2.  dashed
3.  dotted
4.  double
5.  hidden
6.  none

The ListItemType consist of the following key value pairs:

| Key               | Type          | Description                                                         |
| ----------------- | ------------- | ------------------------------------------------------------------- |
| id                | string/number | The id of the list element                                          |
| listLabel         | string        | The list label                                                      |
| isAnchor          | boolean       | Flag to determine whether or not to render the content as an anchor |
| isButton          | boolean       | Flag to determine whether or not to render the content as a button  |
| href              | string        | The href for the anchor element                                     |
| leftSlot          | Slot          | Slot for the left content                                           |
| rightSlot         | Slot          | SLot for the right content                                          |
| btnClickHandler   | Function      | The handler to be called when the button is clicked                 |
| listItemPartName  | string        | The part name for the list item                                     |
| contentPartName   | string        | The part name for the list item content                             |
| listLabelPartName | string        | The part name for the list item label                               |

The Slot type consist of the following key value pairs:

| Key      | Type    | Description                                         |
| -------- | ------- | --------------------------------------------------- |
| hasSlot  | boolean | Flag to determine whether or not to render the slot |
| slotName | string  | The name of the slot to render                      |

# Examples

Ordered List default

export const orderedListTemplate = ({ listType, listItems }) =>
  html`<list-component .listType=${listType} .listItems=${listItems} />`;

<Canvas>
  <Story name="ordered list" args={orderedListVariant}>
    {orderedListTemplate.bind({})}
  </Story>
</Canvas>

Unordered List default

export const unOrderedListTemplate = ({ listType, listItems }) =>
  html`<list-component .listType=${listType} .listItems=${listItems} />`;

<Canvas>
  <Story name="unordered list" args={unOrderedListVariant}>
    {unOrderedListTemplate.bind({})}
  </Story>
</Canvas>

Unordered List with default list styles overridden and bottom border

export const unOrderedListWithDefaultOverriddenAndBorderBottomTemplate = ({
  listType,
  listItems,
  showBorder,
  borderPosition,
  borderStyle,
  overrideDefaultListStyles,
  leftSlot,
}) =>
  html`<list-component
    .listType=${listType}
    .listItems=${listItems}
    .overrideDefaultListStyles=${overrideDefaultListStyles}
    .showBorder=${showBorder}
    .borderPosition=${borderPosition}
    .borderStyle=${borderStyle}
    .leftSlot=${leftSlot}
  >
    <div slot="list-slot">
      <img
        src="https://static.thenounproject.com/png/1916331-200.png"
        alt="caret"
        width="24px"
        height="24px"
      />
    </div>
    <div slot="list-slot-1">
      <img
        src="https://static.thenounproject.com/png/1916331-200.png"
        alt="caret"
        width="24px"
        height="24px"
      />
    </div>
  </list-component>`;

<Canvas>
  <Story
    name="unordered list with default list styles overridden and border bottom"
    args={unOrderedListWithBottomBorder}
  >
    {unOrderedListWithDefaultOverriddenAndBorderBottomTemplate.bind({})}
  </Story>
</Canvas>
