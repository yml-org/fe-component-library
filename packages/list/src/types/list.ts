export type ListType = 'Ordered' | 'Unordered';

export type BorderPosition = 'All' | 'Top' | 'Left' | 'Bottom' | 'Right';

export type BorderStyle =
  | 'solid'
  | 'dashed'
  | 'dotted'
  | 'double'
  | 'hidden'
  | 'none';

export type Slot = {
  slotName?: string;
  hasSlot?: boolean;
};

export type ListItemType = {
  id?: string | number;
  listLabel?: string;
  isAnchor?: boolean;
  isButton?: boolean;
  href?: string;
  rightSlot?: Slot;
  leftSlot?: Slot;
  btnClickHandler?: (id: string | number, listLabel: string) => void;
  listItemPartName?: string;
  contentPartName?: string;
  listLabelPartName?: string;
};

export type ListVariant = {
  listType?: ListType;
  listItems: ListItemType[];
  overrideDefaultListStyles?: boolean;
  showBorder?: boolean;
  borderPosition?: BorderPosition;
  borderStyle?: BorderStyle;
  containerPartName?: string;
  listPartName?: string;
};
