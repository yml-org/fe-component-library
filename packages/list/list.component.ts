import { ListComponent } from './src/list/list.component';

declare global {
  interface HTMLElementTagNameMap {
    'ymlwebcl-list': ListComponent;
  }
}
