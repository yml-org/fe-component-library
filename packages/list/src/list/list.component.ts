import { html, nothing } from 'lit';
import { customElement, property } from 'lit/decorators.js';
import { repeat } from 'lit/directives/repeat.js';
import { TailwindElement } from '../shared/tailwind.element';
import { msg, str } from '@lit/localize';
import {
  ListType,
  ListItemType,
  BorderPosition,
  BorderStyle,
} from '../types/list';
import { ListTypes, BorderPositions, BorderStyles } from '../constants/list';

@customElement('list-component')
export class ListComponent extends TailwindElement(null) {
  @property()
  listType?: ListType = ListTypes.Ordered;
  @property()
  listItems?: ListItemType[] = [];
  @property()
  overrideDefaultListStyles?: boolean = false;
  @property()
  showBorder?: boolean = false;
  @property()
  borderPosition?: BorderPosition = BorderPositions.All;
  @property()
  borderStyle?: BorderStyle = BorderStyles.Solid;
  @property()
  containerPartName?: string = 'webcl-list-container';
  @property()
  listPartName?: string = 'webcl-list';

  protected renderListItemContent(rightSlot, leftSlot, listLabel) {
    return html` ${leftSlot?.hasSlot
        ? html`<slot name=${leftSlot?.slotName} />`
        : nothing}
      <span>${msg(str`${listLabel}`)}</span>
      ${rightSlot?.hasSlot
        ? html`<slot name=${rightSlot?.slotName} />`
        : nothing}`;
  }

  protected renderAnchorOrButton({
    id,
    isAnchor,
    isButton,
    href,
    listLabel,
    rightSlot,
    leftSlot,
    btnClickHandler,
  }: ListItemType) {
    return isAnchor
      ? html`<a href=${href} class="w-full flex basis-full flex-row"
          >${this.renderListItemContent(rightSlot, leftSlot, listLabel)}
        </a>`
      : isButton
      ? html`<button
          @click=${() => btnClickHandler(id, listLabel)}
          class="w-full flex basis-full flex-row"
        >
          ${this.renderListItemContent(rightSlot, leftSlot, listLabel)}
        </button>`
      : html`<span
          >${this.renderListItemContent(rightSlot, leftSlot, listLabel)}</span
        >`;
  }

  protected renderBorderStyles() {
    const { borderPosition, borderStyle } = this;
    switch (borderPosition) {
      case BorderPositions.Bottom:
        return `border-${borderStyle} border-b`;
      case BorderPositions.Top:
        return `border-${borderStyle} border-t`;
      case BorderPositions.Right:
        return `border-${borderStyle} border-r`;
      case BorderPositions.Left:
        return `border-${borderStyle} border-l`;
      case BorderPositions.All:
      default:
        return `border border-${borderStyle}`;
    }
  }

  protected renderListItems() {
    const { showBorder, renderBorderStyles } = this;
    return html` ${repeat(
      this.listItems,
      (listItem) => listItem?.id || listItem?.listLabel,
      (listItem) =>
        html`
          <li
            class=${showBorder ? renderBorderStyles.call(this) : ''}
            part=${listItem?.listItemPartName || 'webcl-list-item'}
          >
            ${this.renderAnchorOrButton(listItem)}
          </li>
        `
    )}`;
  }

  protected renderUnorderedList() {
    const { overrideDefaultListStyles } = this;
    return html`<ul
      part=${this.listPartName}
      class=" flex flex-col ${overrideDefaultListStyles ? '' : 'list-disc'}"
    >
      ${this.renderListItems()}
    </ul>`;
  }

  protected renderOrderedList() {
    const { overrideDefaultListStyles } = this;
    return html`<ol
      part=${this.listPartName}
      class="flex flex-col ${overrideDefaultListStyles ? '' : 'list-decimal'}"
    >
      ${this.renderListItems()}
    </ol>`;
  }

  protected renderList() {
    const { listType } = this;
    switch (listType) {
      case ListTypes.Unordered:
        return this.renderUnorderedList();
      case ListTypes.Ordered:
      default:
        return this.renderOrderedList();
    }
  }
  render() {
    return html`<div class="webcl-list-wrapper" part=${this.containerPartName}>
      ${this.renderList()}
    </div>`;
  }
}

declare global {
  interface HTMLElementTagNameMap {
    'list-component': ListComponent;
  }
}
