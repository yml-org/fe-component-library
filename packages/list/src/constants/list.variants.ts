import { ListItemType, ListVariant } from '../types/list';
import {
  BorderPositions,
  BorderStyles,
  ListTypes,
  SlotPositions,
} from './list';

const listItemsMock: ListItemType[] = [
  {
    id: 1,
    listLabel: 'Account Details',
  },
  {
    id: 2,
    listLabel: 'Notifications',
  },
  {
    id: 3,
    listLabel: 'About',
    isAnchor: true,
    href: 'http://google.com/',
    hasSlot: true,
    slotName: 'list-slot',
    slotPosition: SlotPositions.End,
  },
  {
    id: 4,
    listLabel: 'Technical Support',
  },
  {
    id: 5,
    listLabel: 'Logout',
    isButton: true,
    btnClickHandler: (id, listLabel) => {
      console.log(id, listLabel);
    },
    hasSlot: true,
    slotName: 'list-slot-1',
    slotPosition: SlotPositions.End,
  },
];

const orderedListVariant: ListVariant = {
  listType: ListTypes.Ordered,
  listItems: listItemsMock,
};

const unOrderedListVariant: ListVariant = {
  listType: ListTypes.Unordered,
  listItems: listItemsMock,
};

const unOrderedListWithBottomBorder: ListVariant = {
  ...unOrderedListVariant,
  showBorder: true,
  borderPosition: BorderPositions.Bottom,
  borderStyle: BorderStyles.Solid,
  overrideDefaultListStyles: true,
};

export {
  orderedListVariant,
  unOrderedListVariant,
  unOrderedListWithBottomBorder,
};
