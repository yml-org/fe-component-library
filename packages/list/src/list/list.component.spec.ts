import { ListTypes } from '../constants/list';
import { LitElement } from 'lit-element';
import { listItemsMock } from '../constants/list.variants';

describe('avatar-component', () => {
  const LIST_COMPONENT = 'list-component';
  let listElement: LitElement;

  const getShadowRoot = (tagName: string): ShadowRoot =>
    document.body.getElementsByTagName(tagName)[0].shadowRoot;

  beforeEach(() => {
    listElement = window.document.createElement(LIST_COMPONENT) as LitElement;
    document.body.appendChild(listElement);
  });

  afterEach(() => {
    document.body.getElementsByTagName(LIST_COMPONENT)[0].remove();
  });

  it('matches component snapshot', async () => {
    expect(
      document.body.getElementsByTagName(LIST_COMPONENT)[0]
    ).toMatchSnapshot();
  });
  it('renders the list tag properly based on the listType property', async () => {
    listElement['listType'] = ListTypes.Unordered;
    await listElement.updateComplete;
    const element = getShadowRoot(LIST_COMPONENT);
    expect(element.querySelectorAll('ul').length).toBeTruthy();
  });
  it('renders the list elements correctly', async () => {
    listElement['listItems'] = listItemsMock;
    await listElement.updateComplete;
    const element = getShadowRoot(LIST_COMPONENT);
    expect(element.querySelectorAll('li').length).toEqual(listItemsMock.length);
  });
  it('renders the button element if isButton is set to true for list item', async () => {
    listElement['listItems'] = listItemsMock;
    await listElement.updateComplete;
    const element = getShadowRoot(LIST_COMPONENT);
    expect(element.querySelectorAll('button').length).toEqual(1);
  });
  it('handles the click event on the button', async () => {
    const mockFn = jest.fn();
    const listItems = [
      ...listItemsMock,
      {
        id: 6,
        listLabel: 'Test',
        isButton: true,
        btnClickHandler: mockFn,
      },
    ];
    listElement['listItems'] = listItems;
    await listElement.updateComplete;
    const btnElement =
      getShadowRoot(LIST_COMPONENT).querySelectorAll('button')[1];
    btnElement?.click();
    expect(mockFn).toBeCalled();
  });
  it('sets the href for the anchor tag', async () => {
    listElement['listItems'] = listItemsMock;
    await listElement.updateComplete;
    const anchorElementHref = getShadowRoot(LIST_COMPONENT)
      .querySelector('a')
      .getAttribute('href');

    expect(anchorElementHref).toBe(
      listItemsMock?.find((item) => item?.isAnchor)?.href
    );
  });
});
