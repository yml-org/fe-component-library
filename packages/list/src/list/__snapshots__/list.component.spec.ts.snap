// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`avatar-component matches component snapshot 1`] = `
_ {
  "_$AL": Map {},
  "_$Do": T {
    "_$AA": <!---->,
    "_$AB": null,
    "_$AH": Et {
      "_$AD": R {
        "el": <template />,
        "parts": Array [
          Object {
            "index": 1,
            "type": 2,
          },
        ],
      },
      "_$AM": [Circular],
      "_$AN": undefined,
      "u": Array [
        T {
          "_$AA": <!--?lit$494607496$-->,
          "_$AB": 
    ,
          "_$AH": Et {
            "_$AD": R {
              "el": <template />,
              "parts": Array [
                Object {
                  "ctor": [Function],
                  "index": 0,
                  "name": "class",
                  "strings": Array [
                    "flex flex-col ",
                    "",
                  ],
                  "type": 1,
                },
                Object {
                  "index": 1,
                  "type": 2,
                },
              ],
            },
            "_$AM": [Circular],
            "_$AN": undefined,
            "u": Array [
              j {
                "_$AH": Array [
                  "list-decimal",
                ],
                "_$AM": [Circular],
                "_$AN": undefined,
                "element": <ol
                  class="flex flex-col list-decimal"
                  part="webcl-list"
                >
                  
      
                  <!--?lit$494607496$-->
                   
                  <!--?lit$494607496$-->
                  <!--?-->
                  
    
                </ol>,
                "name": "class",
                "options": Object {
                  "host": [Circular],
                  "isConnected": true,
                  "renderBefore": null,
                },
                "strings": Array [
                  "flex flex-col ",
                  "",
                ],
                "type": 1,
              },
              T {
                "_$AA": <!--?lit$494607496$-->,
                "_$AB": 
    ,
                "_$AH": Et {
                  "_$AD": R {
                    "el": <template />,
                    "parts": Array [
                      Object {
                        "index": 0,
                        "type": 2,
                      },
                    ],
                  },
                  "_$AM": [Circular],
                  "_$AN": undefined,
                  "u": Array [
                    T {
                      "_$AA": <!--?lit$494607496$-->,
                      "_$AB": <!--?-->,
                      "_$AH": Array [],
                      "_$AM": [Circular],
                      "_$AN": undefined,
                      "_$Cl": Object {
                        "_$AM": [Circular],
                        "_$Ci": undefined,
                        "_$Ct": [Circular],
                        "ut": Array [],
                      },
                      "_$Cm": true,
                      "options": Object {
                        "host": [Circular],
                        "isConnected": true,
                        "renderBefore": null,
                      },
                      "type": 2,
                    },
                  ],
                },
                "_$AM": [Circular],
                "_$AN": undefined,
                "_$Cm": true,
                "options": Object {
                  "host": [Circular],
                  "isConnected": true,
                  "renderBefore": null,
                },
                "type": 2,
              },
            ],
          },
          "_$AM": [Circular],
          "_$AN": undefined,
          "_$Cm": true,
          "options": Object {
            "host": [Circular],
            "isConnected": true,
            "renderBefore": null,
          },
          "type": 2,
        },
      ],
    },
    "_$AM": undefined,
    "_$AN": undefined,
    "_$Cm": true,
    "options": Object {
      "host": [Circular],
      "isConnected": true,
      "renderBefore": null,
    },
    "type": 2,
  },
  "_$E_": Promise {},
  "_$Ei": undefined,
  "_$El": null,
  "__borderPosition": "All",
  "__borderStyle": "solid",
  "__listItems": Array [],
  "__listType": "Ordered",
  "__overrideDefaultListStyles": false,
  "__showBorder": false,
  "enableUpdating": [Function],
  "hasUpdated": true,
  "isUpdatePending": false,
  "renderOptions": Object {
    "host": [Circular],
    "isConnected": true,
    "renderBefore": null,
  },
  "renderRoot": ShadowRoot {
    "_$litPart$": T {
      "_$AA": <!---->,
      "_$AB": null,
      "_$AH": Et {
        "_$AD": R {
          "el": <template />,
          "parts": Array [
            Object {
              "index": 1,
              "type": 2,
            },
          ],
        },
        "_$AM": [Circular],
        "_$AN": undefined,
        "u": Array [
          T {
            "_$AA": <!--?lit$494607496$-->,
            "_$AB": 
    ,
            "_$AH": Et {
              "_$AD": R {
                "el": <template />,
                "parts": Array [
                  Object {
                    "ctor": [Function],
                    "index": 0,
                    "name": "class",
                    "strings": Array [
                      "flex flex-col ",
                      "",
                    ],
                    "type": 1,
                  },
                  Object {
                    "index": 1,
                    "type": 2,
                  },
                ],
              },
              "_$AM": [Circular],
              "_$AN": undefined,
              "u": Array [
                j {
                  "_$AH": Array [
                    "list-decimal",
                  ],
                  "_$AM": [Circular],
                  "_$AN": undefined,
                  "element": <ol
                    class="flex flex-col list-decimal"
                    part="webcl-list"
                  >
                    
      
                    <!--?lit$494607496$-->
                     
                    <!--?lit$494607496$-->
                    <!--?-->
                    
    
                  </ol>,
                  "name": "class",
                  "options": Object {
                    "host": [Circular],
                    "isConnected": true,
                    "renderBefore": null,
                  },
                  "strings": Array [
                    "flex flex-col ",
                    "",
                  ],
                  "type": 1,
                },
                T {
                  "_$AA": <!--?lit$494607496$-->,
                  "_$AB": 
    ,
                  "_$AH": Et {
                    "_$AD": R {
                      "el": <template />,
                      "parts": Array [
                        Object {
                          "index": 0,
                          "type": 2,
                        },
                      ],
                    },
                    "_$AM": [Circular],
                    "_$AN": undefined,
                    "u": Array [
                      T {
                        "_$AA": <!--?lit$494607496$-->,
                        "_$AB": <!--?-->,
                        "_$AH": Array [],
                        "_$AM": [Circular],
                        "_$AN": undefined,
                        "_$Cl": Object {
                          "_$AM": [Circular],
                          "_$Ci": undefined,
                          "_$Ct": [Circular],
                          "ut": Array [],
                        },
                        "_$Cm": true,
                        "options": Object {
                          "host": [Circular],
                          "isConnected": true,
                          "renderBefore": null,
                        },
                        "type": 2,
                      },
                    ],
                  },
                  "_$AM": [Circular],
                  "_$AN": undefined,
                  "_$Cm": true,
                  "options": Object {
                    "host": [Circular],
                    "isConnected": true,
                    "renderBefore": null,
                  },
                  "type": 2,
                },
              ],
            },
            "_$AM": [Circular],
            "_$AN": undefined,
            "_$Cm": true,
            "options": Object {
              "host": [Circular],
              "isConnected": true,
              "renderBefore": null,
            },
            "type": 2,
          },
        ],
      },
      "_$AM": undefined,
      "_$AN": undefined,
      "_$Cm": true,
      "options": Object {
        "host": [Circular],
        "isConnected": true,
        "renderBefore": null,
      },
      "type": 2,
    },
  },
}
`;
